### PR TITLE
Shifted the CallActor to call executeCommand asynchronously

### PR DESCRIPTION
**Pull request by cjllanwarne**
To allow interruptions. Should otherwise operate exactly the same as currently.

### CODE REVIEW

**Comment by cjllanwarne** [comment]
For reviewers:
@scottfrazer 
@mcovarr

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 1]
```diff
@@ -62,11 +65,14 @@ class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backe
...
-          context.parent ! WorkflowActor.CallFailed(call, e.getMessage)
+
+      val callExecution: Future[Try[Map[String, WdlValue]]] = Future {
+        backend.executeCommand(commandLine, workflowDescriptor, call, backendInputs, inputName => locallyQualifiedInputs.get(inputName).get)
+      }
+
+      callExecution onComplete {
+        case Success(result) => receiveResults(result)
+        case Failure(error) => receiveResults(Failure(error))
       }
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/CallActor.scala]: There could be some flattening before processing results:

```scala
      // Win internet points from Jeff and me if you make this a case class.
      type CallOutputs = Map[String, WdlValue]
      // Future and not Future.fromTry required here to make execution truly async.
      val futureResults: Future[Try[CallOutputs]] = Future {
        backend.executeCommand(commandLine, workflowDescriptor, call, backendInputs, inputName => locallyQualifiedInputs.get(inputName).get)
      }

      val futureCall: Future[CallOutputs] = for {
        presentResults <- futureResults
        results <- Future.fromTry(presentResults)
      } yield results

      futureCall onComplete {
        // downstream changes
        ...
```

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 2]
```diff
@@ -75,4 +81,13 @@ class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backe
       case Failure(e) => failedInstantiation(e)
     }
   }
+
+  def receiveResults(results: Try[Map[String, WdlValue]]): Unit = {
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/CallActor.scala]: It feels a bit weird that both branches of the above effectively coalesce to call this and then it splits out success and failure again, perhaps the logic here can be inlined with the above if the previous suggestion is taken.

Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/CallActor.scala]: This turned out to be almost a necessary result of the above comment, but I do need to ask you about that one, I'm not quite getting how it's supposed to work

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 3]
```diff
@@ -8,8 +8,11 @@ import cromwell.engine.backend.Backend
 import cromwell.engine.workflow.WorkflowActor
 import cromwell.engine.workflow.WorkflowActor.CallFailed
 import cromwell.engine
+
+import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/CallActor.scala]: Can someone confirm this is the correct (implicit) execution context to use?

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 4]
```diff
@@ -62,8 +66,18 @@ class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backe
...
       log.info(s"$tag: launching `$commandLine`")
       originalSender ! WorkflowActor.CallStarted(call)
-      backend.executeCommand(commandLine, workflowDescriptor, call, backendInputs, inputName => locallyQualifiedInputs.get(inputName).get) match {
-        case Success(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
+
+      val futureResults: Future[Try[CallOutputs]] = Future {
+        backend.executeCommand(commandLine, workflowDescriptor, call, backendInputs, inputName => locallyQualifiedInputs.get(inputName).get)
+      }
+
+      val futureCall: Future[CallOutputs] = for {
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/CallActor.scala]: twiddle this name, otherwise :+1:

**Pull request merged**